statannotations/Annotator.py

```diff
@@ -216,10 +218,11 @@ def annotate(self, line_offset=None, line_offset_to_group=None):
         self.validate_test_short_name()
 
         for annotation in self.annotations:
-            self._annotate_pair(annotation,
-                                ax_to_data=ax_to_data,
-                                ann_list=ann_list,
-                                orig_value_lim=orig_value_lim)
+            if annotation.check_data_stat_result() and (not self.hide_non_significant or annotation.data.is_significant):
+                self._annotate_pair(annotation,
+                                    ax_to_data=ax_to_data,
+                                    ann_list=ann_list,
+                                    orig_value_lim=orig_value_lim)
```

Comment:
It looks like we'll prevent annotating results other than StatResults here, which is not the point. What happens is that we can't hide them in that case. (And the user just has to not pass them if they don't want to annotate with them).
So instead, I think we need something like this

```suggestion

```
```suggestion
            if self.hide_non_significant and isinstance(self.data, StatResult) \
                    and not annotation.data.is_significant:
                continue
            self._annotate_pair(annotation,
                                ax_to_data=ax_to_data,
                                ann_list=ann_list,
                                orig_value_lim=orig_value_lim)
```

Reply:
I'm unfortunately not familiar enough with all edge cases (like keeping the option to pass other instances than StatResult), though given your explanation that does make sense.